### PR TITLE
(PDB-1998) use a coalesce instead of jsonb migration

### PR DESCRIPTION
**Pull request by wkalt**
This changes our jsonb migration to leave the existing json column in place,
and changes our query code to use a coalesce of the two instead of the one-shot
migration to jsonb.

### CODE REVIEW

File: src/puppetlabs/puppetdb/query_eng/engine.clj [hunk 1]
```diff
@@ -359,12 +365,17 @@
       "metrics" {:type :json
                  :queryable? false
                  :field {:select [(h/row-to-json :t)]
-                         :from [[{:select [[:metrics :data] [(hsql-hash-as-str :hash) :href]]} :t]]}
+                         :from [[{:select
+                                  [[(h/coalesce :metrics
+                                                (h/scast :metrics_json (jsonb-type))) :data]
```
Comment (wkalt) [inline src/puppetlabs/puppetdb/query_eng/engine.clj]: still testing this with a large dataset. my remaining concern is that this cast could be very expensive for queries returning large numbers of reports.

File: src/puppetlabs/puppetdb/query_eng/engine.clj [hunk 2]
```diff
@@ -297,7 +298,9 @@
   []
   (map->Query {:projections {"logs" {:type :json
                                      :queryable? false
-                                     :field :reports.logs}
+                                     :field (h/coalesce :logs
```
Comment (senior) [inline src/puppetlabs/puppetdb/query_eng/engine.clj]: For HyperSQL I guess this hinges on there always being a value in the logs column? What happens in HyperSQL if logs in null?

Comment (senior) [inline src/puppetlabs/puppetdb/query_eng/engine.clj]: I think we also need tests around the various scenarios here (data in JSON, data in JSONB etc)

Comment (wkalt) [inline src/puppetlabs/puppetdb/query_eng/engine.clj]: @senior added a test to cover these cases

**Pull request merged**